os/compression/compress_read.c

```diff
@@ -237,7 +237,7 @@ static off_t compress_lseek_block(int filfd, uint16_t binary_header_size, int bl
 static off_t compress_read_block(int filfd, uint16_t binary_header_size, FAR uint8_t *buf, int block_number)
 {
 	off_t rpos;
-	size_t readsize;
+	off_t readsize;
```

Comment:
negative value is acceptable ? if not, you should use size_t or something instead of off_t.
Of course return type of compress_read_block also need to be changed